### PR TITLE
[ROCm] enable tests test_sampled_addmm_autograd_cuda_*, test_sample…

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
These tests PASS on ROCM 5.6+ now:

- test_sampled_addmm_autograd_cuda_complex128
- test_sampled_addmm_autograd_cuda_complex64
- test_sampled_addmm_autograd_cuda_float32
- test_sampled_addmm_autograd_cuda_float64
- test_sampled_addmm_cuda_complex128
- test_sampled_addmm_cuda_complex64
- test_sampled_addmm_cuda_float32
- test_sampled_addmm_cuda_float64
- test_autograd_dense_output_addmm_cuda_float64
- test_autograd_dense_output_addmv_cuda_float64
- test_autograd_dense_output_mv_cuda_float64

@pruthvistony @jithunnair-amd 

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Comment by dnikolaev-amd** [comment]
@jeffdaily

File: test/test_sparse_csr.py [hunk 1]
```diff
@@ -2772,9 +2770,8 @@ def fn(input):
             dense_output.backward(dense_covector)
             self.assertEqual(sparse_input.grad, dense_input.grad)
 
-    @skipCUDAIfRocm
     @skipCUDAIf(
-        not _check_cusparse_sddmm_available(),
+        not (TEST_WITH_ROCM or _check_cusparse_sddmm_available()),
```
Comment (pruthvistony) [inline test/test_sparse_csr.py:2774]: Can the function - _check_cusparse_sddmm_available() updated to handle the TEST_WITH_ROCM condition.

Comment (dnikolaev-amd) [reply test/test_sparse_csr.py]: TEST_WITH_ROCM moved into _check_cusparse_sddmm_available() in the next commit

**Comment by dnikolaev-amd** [comment]
Looks like test_sampled_addmm_zero_sized_cuda_* passes and can be enabled too https://github.com/pytorch/pytorch/blob/1462d72904cb81917b9355d6a58916f389e9084c/test/test_sparse_csr.py#L2480

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@malfet 
Can you help review this PR.

**Review by jeffdaily — APPROVED** [review]

**Comment by dnikolaev-amd** [comment]
@kit1980 
Can you help review this PR?

**Comment by jeffdaily** [comment]
@atalman @huydhn adding as reviewers for adding back ROCm UTs.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**